Source/Core/Core/DSP/DSPAssembler.cpp

```diff
@@ -71,32 +68,18 @@ bool DSPAssembler::Assemble(const std::string& text, std::vector<u16>& code,
     return false;
 
   // We now have the size of the output buffer
-  if (m_totalSize > 0)
-  {
-    gdg_buffer = (char*)malloc(m_totalSize * sizeof(u16) + 4);
-    if (!gdg_buffer)
-      return false;
-
-    memset(gdg_buffer, 0, m_totalSize * sizeof(u16));
-  }
-  else
+  if (m_totalSize <= 0)
     return false;
 
+  m_output_buffer.resize(m_totalSize);
```

Comment:
Intentional. m_totalSize tracks the size of all opcodes being put in the output buffer